app/controllers/candidate_interface/sign_in_controller.rb

```diff
@@ -12,24 +12,27 @@ def create
       SignInCandidate.new(candidate_params[:email_address], self).call
     end
 
+    def confirm_authentication
+      candidate = retrieve_candidate_by_token || retrieve_candidate_by_user_id
+      redirect_to action: :new if candidate.nil?
+    end
+
     def authenticate
-      candidate = FindCandidateByToken.call(raw_token: params[:token])
+      candidate = retrieve_candidate_by_token
       token_not_expired = FindCandidateByToken.token_not_expired?(candidate)
 
-      if candidate.nil? && params[:u]
-        candidate_id = Encryptor.decrypt(params[:u])
-        candidate = Candidate.find(candidate_id) if candidate_id
+      if candidate.nil?
+        candidate = retrieve_candidate_by_user_id
       end
 
-      if candidate.nil?
-        redirect_to action: :new
-      elsif token_not_expired
+      redirect_to(action: :new) and return if candidate.nil?
+
+      if token_not_expired
         flash[:success] = t('apply_from_find.account_created_message') if candidate.last_signed_in_at.nil?
         sign_in(candidate, scope: :candidate)
         add_identity_to_log candidate.id
-        candidate.update!(last_signed_in_at: Time.zone.now)
-
-        redirect_to candidate_interface_interstitial_path(providerCode: params[:providerCode], courseCode: params[:courseCode])
```

Comment:
Is there another route that accepts these params?

Reply:
Appears to be handled by both SignUpController and ApplyFromFindController.

See:
https://github.com/DFE-Digital/apply-for-teacher-training/blob/1936-spike-auth-after-button-press/spec/system/candidate_interface/course_selection/candidate_new_user_with_course_params_spec.rb
https://github.com/DFE-Digital/apply-for-teacher-training/blob/1936-spike-auth-after-button-press/spec/system/candidate_interface/course_selection/candidate_existing_user_with_course_params_spec.rb
https://github.com/DFE-Digital/apply-for-teacher-training/blob/1936-spike-auth-after-button-press/spec/system/candidate_interface/course_selection/candidate_signed_in_user_with_course_params_spec.rb